### PR TITLE
Feature/placed pages pagination

### PR DESCRIPTION
**Pull request by danielebogo**
This PR allows the pages list to load all the pages till the end, without using the infinite scroll feature as before.

Page List has a new `WPContentSyncHelper` called `PageContentSyncHelper` in charge to recursively call `syncMoreContent` till it has no more content.

**To test:**
- Open a blog and check if all the pages are loaded

@aerych do you mind to have a look since we already spoke about this?

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Pages/PageListViewController.swift [hunk 1]
```diff
@@ -45,6 +45,20 @@ class PageListViewController: AbstractPostListViewController, UIViewControllerRe
         }
     }
 
+    @objc private lazy var _syncHelper: PageContentSyncHelper = {
+        let syncHelper = PageContentSyncHelper()
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Pages/PageListViewController.swift]: I think the change I suggested above would avoid the need for this or for customizing the getter/setter below.

**Comment by danielebogo** [comment]
Thanks @aerych for the hint. Yes you are right, the freeze happened because of the purgeLocalSync that was applied for every batch.
So the fix now affect only the `PostService`. The syncPost method use a recursion to load and sync/save the remote posts at the end. In this way the sync delegate use one function only and doesn't call the load more anymore (that actually is a duplication too).
Let me know what you think 😉!

File: WordPress/Classes/Services/PostService.m [hunk 2]
```diff
@@ -123,33 +123,66 @@ - (void)syncPostsOfType:(PostServiceType)postType
                 forBlog:(Blog *)blog
                 success:(PostServiceSyncSuccess)success
                 failure:(PostServiceSyncFailure)failure
+{
+    [self syncPostsOfType:postType
+              withOptions:options
+                  forBlog:blog
+              loadedPosts:@[].mutableCopy
```
Comment (aerych) [inline WordPress/Classes/Services/PostService.m]: This is super nitpicky.  Our [objc style guide](https://github.com/wordpress-mobile/objective-c-style-guide) calls for dot notation only for `@proprties`.  Method calls in particular we like to retain the bracket syntax: `[@[] mutableCopy]`.

File: WordPress/Classes/Services/PostService.m [hunk 3]
```diff
@@ -123,33 +123,66 @@ - (void)syncPostsOfType:(PostServiceType)postType
...
+                                         failure:failure];
+                       } else {
+                           [self.managedObjectContext performBlock:^{
+                               NSError *error;
+                               Blog *blogInContext = (Blog *)[self.managedObjectContext existingObjectWithID:blogObjectID error:&error];
+                               if (!blogInContext || error) {
+                                   DDLogError(@"Could not retrieve blog in context %@", (error ? [NSString stringWithFormat:@"with error: %@", error] : @""));
+                                   return;
+                               }
+                               [self mergePosts:loadedPosts.copy
```
Comment (aerych) [inline WordPress/Classes/Services/PostService.m]: And then also bracket notation here `[loadedPosts copy]`

**Comment by danielebogo** [comment]
Thanks @aerych ! I fixed the brackets syntax and I tested it with the ~2500 pages blog! Everything works fine!

**Pull request merged**